capstone-frontend/src/components/ClubPage.jsx

```diff
@@ -125,6 +125,25 @@ class ClubPage extends Component {
       });
   }
 
+  handleMemberChange = (member, type) => {
+    if (type === 'add') {
+
+      const club = this.state.club;
+      club.memberIDs.push(member.id);
+      
+      this.setState({ members: [...this.state.members, member], club });
+    }
+    else if (type === 'remove') {
+
+      const members = this.state.members.filter(knownMember => knownMember.id !== member.id);
+
+      const club = this.state.club;
+      club.memberIDs = club.memberIDs.filter(memberId => memberId !== member.id);
```

Comment:
Good use of `filter` to avoid mutating `memberIDs`, but this still mutates `this.state.club`. You could instead make create a `memberIDs` constant here to store the result of the `filter` then set the new state to `club: {...club, memberIDs}`.